src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -2838,6 +2838,176 @@ public async Task DebugView_Endpoints_UseEndpoints_AvailableBeforeAndAfterStart(
+        });
+
+        app.UseRouting();
+
+        await app.StartAsync();
+
+        var client = app.GetTestClient();
+        var response = await client.GetAsync("/?username=test");
+        response.EnsureSuccessStatusCode();
+        Assert.Equal("Hello test!", await response.Content.ReadAsStringAsync());
+    }
+
+    [Fact]
+    public async Task ImplicitMiddlewares_RunBeforeExplicitRouting_TerminalMiddleware_AfterUseRouting()
+    {
+        var builder = WebApplication.CreateBuilder();
+        builder.WebHost.UseTestServer();
+        builder.Services.AddAuthentication("testSchemeName")
+            .AddScheme<AuthenticationSchemeOptions, UberHandler>("testSchemeName", "testDisplayName", _ => { });
+        builder.Services.AddAuthorization();
+        await using var app = builder.Build();
+
+        app.UseRouting();
+
+        app.Run((HttpContext context) =>
+        {
+            Assert.NotNull(context.Features.Get<IAuthenticationFeature>());
+            return context.Response.WriteAsync($"Hello {context.Request.Query["username"]}!");
+        });
+
+        await app.StartAsync();
+
+        var client = app.GetTestClient();
+        var response = await client.GetAsync("/?username=test");
+        response.EnsureSuccessStatusCode();
+        Assert.Equal("Hello test!", await response.Content.ReadAsStringAsync());
+    }
+
+    [Fact]
+    public async Task ImplicitMiddlewares_RunBetween_ExplicitRouting_TerminalMiddlewareMapGet()
+    {
+        var builder = WebApplication.CreateBuilder();
+        builder.WebHost.UseTestServer();
+        builder.Services.AddAuthentication("testSchemeName")
+            .AddScheme<AuthenticationSchemeOptions, UberHandler>("testSchemeName", "testDisplayName", _ => { });
+        builder.Services.AddAuthorization();
+        await using var app = builder.Build();
+
+        app.UseRouting();
+
+        app.Run((HttpContext context) =>
+        {
+            Assert.NotNull(context.Features.Get<IAuthenticationFeature>());
+            return context.Response.WriteAsync($"Hello {context.Request.Query["username"]}!");
+        });
+
+        app.MapGet("/endpoint", (HttpContext context, string username) =>
+        {
+            Assert.NotNull(context.Items["__AuthorizationMiddlewareWithEndpointInvoked"]);
+            Assert.NotNull(context.Features.Get<IAuthenticationFeature>());
+            return $"Hello {username}!";
+        }).AllowAnonymous();
+
+        await app.StartAsync();
+
+        var client = app.GetTestClient();
+        var response = await client.GetAsync("/?username=test");
+        response.EnsureSuccessStatusCode();
+        Assert.Equal("Hello test!", await response.Content.ReadAsStringAsync());
+
+        var endpointResponse = await client.GetAsync("/endpoint?username=test");
+        endpointResponse.EnsureSuccessStatusCode();
+        Assert.Equal("Hello test!", await endpointResponse.Content.ReadAsStringAsync());
```

Comment:
nit: since the 2 endpoints output that same thing you can't tell if they both ran